src/client/runner/runner-base.js

```diff
@@ -373,18 +371,17 @@ RunnerBase.prototype._runInIFrame = function (iframe, stepName, step, stepNum) {
         }
     }
 
-    pingIframe(iframe, function (err) {
-        if (err) {
+    pingIframe(iframe)
+        .then(() => {
+            runner.iframeExistenceWatcherInterval = window.setInterval(iframeExistenceWatcher, IFRAME_EXISTENCE_WATCHING_INTERVAL);
+            messageSandbox.sendServiceMsg(msg, iframe.contentWindow);
+        })
+        .catch(() => {
```

Comment:
It is wrong.
Need to rewrite without `catch`.
see commit https://github.com/miherlosev/testcafe-hammerhead/commit/28757401aaea8709544bd150eb50e83887758026